### PR TITLE
Remove compiler warning if only MBEDTLS_PK_PARSE_C is defined

### PR DESCRIPTION
**Pull request by kennethsoerensen**
Warning reported with IAR compiler:
"mbedtls\library\pkparse.c",1167  Warning[Pe550]: variable "ret" was set but never used

Backports: https://github.com/ARMmbed/mbedtls/pull/4903, https://github.com/ARMmbed/mbedtls/pull/4904

### CODE REVIEW

**Comment by RonEld** [comment]
@kennethsoerensen Thank you for your contribution!
Please confirm [this](https://os.mbed.com/users/kennethsoerensen/) is your Mbed account

**Comment by RonEld** [comment]
Note to reviewers:
ALthough this is a compiler warning, that should not happen, it will probably be fixed with the merge of #2124 because in https://github.com/ARMmbed/mbedtls/pull/2124/files#diff-194f3e5cff8bbce913f1a6e2ed06afffR1421 there is a usage of `ret`.
I suggest waiting until it will be merged

**Comment by kennethsoerensen** [comment]
> 
> 
> @kennethsoerensen Thank you for your contribution!
> Please confirm [this](https://os.mbed.com/users/kennethsoerensen/) is your Mbed account

Yes, it is

**Review by gilles-peskine-arm — APPROVED** [review]

**Comment by gilles-peskine-arm** [comment]
Thanks for updating this, and sorry it fell through the cracks for so long!

Would you mind making similar pull requests for the other supported-branches (`development_2.x` and `mbedtls-2.16`)?

**Comment by kennethsoerensen** [comment]
Backports created - please see #4903 and #4904

**Pull request merged**
